src/stream/src/common/table/state_table.rs

```diff
@@ -138,13 +142,23 @@ pub struct StateTableInner<
 
     /// Watermark cache
     watermark_cache: StateTableWatermarkCache,
+
+    /// Data Types
+    /// We will need to use to build data chunks from state table rows.
+    data_types: Vec<DataType>,
```

Comment:
I guess it should be accessible within the deserializer from `SD`?

Reply:
Yes but seems a little strange to expose it via the deserializer.